### ISSUE TITLE
Is it possible to use CMAKE to handle project ,so source code can be compiled on windows?

### ISSUE THREAD

**Issue by cathty** [stevengj/meep#385]
Is it possible to use CMAKE to handle project ,so source code can be compiled on windows?

**Comment by stevengj** [comment]
It wouldn’t really help, since you’d also need to compile all the dependencies. 

The best way to compile on Windows is currently to install Cygwin.

**Comment by cenit** [comment]
I'd like to be able to build meep on Windows "natively", exploiting [vcpkg](https://github.com/Microsoft/vcpkg) for external libraries (blas, lapack, mpi, hdf5, ...), starting from a portable CMakeLists.txt for this code. If I understand correctly, to have only the C++ interface and barely no special output format, the dependencies are even zero, am I right? So that the work can proceed step-by-step, and starting with a barely functional meep with the minimum effort, then expanding to all dependencies. 
Is there any other one interested, to split the work?

**Comment by stevengj** [comment]
I'd rather support Windows via Conda packages, since that's how we are currently distributing Meep binaries.

**Comment by cenit** [comment]
but you still have to build it via Visual Studio, am I wrong?

**Comment by stevengj** [comment]
No, we can use mingw, which is nice because it should allow us to use the existing unixy build system: https://github.com/conda-forge/conda-forge.github.io/issues/112

**Comment by cenit** [comment]
I've lost track of the `MinGW` project long ago, it was really hacky and unsuitable, at least for what I was doing. I will give it a second try, I just finished reading all the thread and it seems that things improved, if Conda-forge has adopted it for many wheels for Windows. I'd like to see this project to run on windows natively asap, and maybe I will keep an eye on a cmake toolchain in the meantime. I hope it won't be too hard (some dependencies like `guile` and `libctl` look more difficult than `meep` itself)!

**Comment by cenit** [comment]
Ok, not my first priority but this is where I started some work in order to build meep (libctl for now ;) ) with mingw on Windows: https://github.com/Microsoft/vcpkg/pull/4997/files
